packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -66,6 +67,14 @@ enum RoutePopDisposition {
   bubble,
 }
 
+/// Name for the method which is called when the system wants the current route
+/// to be shifted to the previous route in the history.
+const String _popRouteMethod = 'popRoute';
+
+/// Name for the method which is called when the system wants to open a
+/// particular page.
+const String _pushRouteMethod = 'pushRoute';
```

Comment:
Since these are sent from the framework to the engine as "notification" that a route has been pushed or popped, maybe a better name would be "routePushed" and "routePopped"?

Also, `pushRoute` and `popRoute` are already being used to send messages from the engine to the framework, so let's rename the methods to avoid any confusion.

And for the sake of completeness, what do you think of having a separate "routeReplaced" method?

Reply:
Done.